pkg/data/helpers.go

```diff
@@ -0,0 +1,39 @@
+package data
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/service/dynamodb/expression"
+)
+
+func getFiltersFromStruct(i interface{}, keyName *string) (*expression.KeyConditionBuilder, *expression.ConditionBuilder) {
```

Comment:
Build filters and key condition (filters).  Using the keyName to determine if a column is part of an index and switch it from a filter to a key condition.